pkg/graph/parser/parser.go

```diff
@@ -80,40 +80,45 @@ func validateSchema(schema *spec.Schema, path string) error {
 	return nil
 }
 
-func getExpectedType(schema *spec.Schema, resource interface{}) (string, error) {
+func getExpectedTypes(schema *spec.Schema, resource interface{}) ([]string, error) {
 	// handle "x-kubernetes-int-or-string"
 	if ext, ok := schema.VendorExtensible.Extensions[xKubernetesIntOrString]; ok {
 		enabled, ok := ext.(bool)
 		if !ok {
-			return "", fmt.Errorf("xKubernetesIntOrString extension is not a boolean")
+			return []string{""}, fmt.Errorf("xKubernetesIntOrString extension is not a boolean")
 		}
 		if enabled {
-			switch resourceType := resource.(type) {
-			case string:
-				return "string", nil
-			case int:
-				return "integer", nil
-			default:
-				return "", fmt.Errorf("found `xKubernetesIntOrString` extension but field value is neither a string nor an integer: %v", resourceType)
-			}
+			return []string{"string", "integer"}, nil
+		}
+		if !enabled {
+			return []string{""}, fmt.Errorf("found `xKubernetesIntOrString` extension but field value is neither a string nor an integer: %T", resource)
 		}
 	}
 
 	if schema.Type[0] != "" {
-		return schema.Type[0], nil
+		return schema.Type, nil
 	}
 	if schema.AdditionalProperties != nil && schema.AdditionalProperties.Allows {
 		// NOTE(a-hilaly): I don't like the type "any", we might want to change this to "object"
 		// in the future; just haven't really thought about it yet.
 		// Basically "any" means that the field can be of any type, and we have to check
 		// the ExpectedSchema field.
-		return schemaTypeAny, nil
+		return []string{schemaTypeAny}, nil
 	}
-	return "", fmt.Errorf("unknown schema type")
+	return []string{""}, fmt.Errorf("unknown schema type")
 }
 
-func parseObject(field map[string]interface{}, schema *spec.Schema, path, expectedType string) ([]variable.FieldDescriptor, error) {
-	if expectedType != "object" && (schema.AdditionalProperties == nil || !schema.AdditionalProperties.Allows) {
+func isExpectedType(expectedTypes []string, expectedType string) bool {
+	for _, exceptedType := range expectedTypes {
+		if exceptedType == expectedType {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I think this function could be named `sliceInclude`  - we can also just use https://pkg.go.dev/slices#Contains